src/main/java/org/opensrp/service/ImportBulkDataService.java

```diff
@@ -0,0 +1,292 @@
+package org.opensrp.service;
+
+import org.opensrp.domain.AssignedLocations;
+import org.opensrp.domain.Organization;
+import org.opensrp.domain.Practitioner;
+import org.opensrp.domain.PractitionerRole;
+import org.opensrp.domain.PractitionerRoleCode;
+import org.opensrp.domain.CodeSystem;
+import org.opensrp.domain.Code;
+import org.opensrp.dto.CsvBulkImportDataSummary;
+import org.opensrp.dto.FailedRecordSummary;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smartregister.domain.PhysicalLocation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class ImportBulkDataService {
+
+	@Autowired
+	private OrganizationService organizationService;
+
+	@Autowired
+	private PractitionerService practitionerService;
+
+	@Autowired
+	private PractitionerRoleService practitionerRoleService;
+
+	@Autowired
+	private PhysicalLocationService physicalLocationService;
+
+	private static Logger logger = LoggerFactory.getLogger(ImportBulkDataService.class.toString());
+
+	private static final String LOCATION_ID_KEY = "Location Id";
+
+	private static final String LOCATION_NAME_KEY = "Location Name";
+
+	private static final String ORGANIZATION_NAME_KEY = "Organization Name";
+
+	private static final String PLAN_ID_KEY = "Plan Id";
+
+	private static final String ORGANIZATION_ID_KEY = "Organization Id";
+
+	private static final String USER_ID_KEY = "User Id";
+
+	private static final String USER_NAME_KEY = "User Name";
+
+	private static final String NAME_KEY = "Name";
+
+	private static final String ROLE_KEY = "Role";
+
+	public CsvBulkImportDataSummary convertandPersistOrganizationdata(List<Map<String, String>> csvOrganizations) {
+
+		CsvBulkImportDataSummary csvBulkImportDataSummary = new CsvBulkImportDataSummary();
+		Integer rowsProcessed = 0;
+		Integer rowsInCsv = csvOrganizations.size();
+		csvBulkImportDataSummary.setNumberOfCsvRows(rowsInCsv);
+		Integer rowCount = 0;
+		List<FailedRecordSummary> failedRecordSummaries = new ArrayList<>();
+		FailedRecordSummary failedRecordSummary;
+
+		Organization organization;
+		String organizationIdentifier;
+		CodeSystem type;
+		List<Code> codes;
+		Code code;
+		Set<AssignedLocations> assignedLocations;
+		AssignedLocations assignedLocation;
+
+		String locationId = "";
+		String planId = "";
+		String organizationName = "";
+		try {
+			for (Map<String, String> csvdata : csvOrganizations) {
+				rowCount++;
+
+				if (validateLocationData(csvdata)) {
```

Comment:
shouldn't you validate the organization here since an organization is being added below. Or check if the Org exists before creating

Reply:
What kind of validation should be added for Organization, as we have a name column only for Organization in the CSV?
As per our discussion earlier, I believe the organization can not be edited. 
To check if an organization exists before, can I find organization by name? Is organization uniquely identified by its name?

Reply:
Check organization by name